### ISSUE TITLE
virtualenv 20.21.1 throwing AttributeError: 'NoneType' object has no attribute 'startswith' on Windows when it's installed in a virtual environment

### ISSUE THREAD

**Issue by thecesrom** [pypa/virtualenv#2589]
**Issue**

I am actively working on Jython packages so I'm stuck with Python 2.7.18, and testing with `tox`, so I had to add the following to my `tox.ini` file:

```ini
[tox]
requires =
    virtualenv<20.22.0
```

I've found that `virtualenv` 20.21.1 fails to create virtual environments when virtualenv itself is installed in a virtual environment and crashes with the following error: `AttributeError: 'NoneType' object has no attribute 'startswith'`

**Environment**

I installed virtualenv 20.21.1 on my base Python installation and created a virtual environment on my HOME directory called `.virtualenv-20.21.1` where I installed version 20.21.1.

Provide at least:

- OS: Windows 10
- `pip list` of the host python where `virtualenv` is installed:

  ```console
  $ python -m pip list
  Package      Version
  ------------ -------
  distlib      0.3.6
  filelock     3.12.2
  pip          23.1.2
  platformdirs 3.5.3
  setuptools   67.8.0
  virtualenv   20.21.1
  wheel        0.40.0
  ```

**Output of the virtual environment creation**

Make sure to run the creation with `-vvv --with-traceback`:

```console
$ which python
/c/Users/thecesrom/.virtualenv-20.21.1/Scripts/python
$ python -m virtualenv -vvv --with-traceback ~/.virtualenv-virtualenv
392 setup logging to NOTSET [DEBUG report:35]
400 find interpreter for spec PythonSpec(path=C:\Users\thecesrom\.virtualenv-20.21.1\Scripts\python.exe) [INFO builtin:56]
401 Attempting to acquire lock 2971795500784 on C:\Users\thecesrom\AppData\Local\pypa\virtualenv\py_info\1\5acf275e529b7f1b869594580444d4303145808c0032da3f6f98a78be5318304.lock [DEBUG _api:212]
402 Lock 2971795500784 acquired on C:\Users\thecesrom\AppData\Local\pypa\virtualenv\py_info\1\5acf275e529b7f1b869594580444d4303145808c0032da3f6f98a78be5318304.lock [DEBUG _api:215]
403 got python info of C:\Users\thecesrom\scoop\apps\coatl310\current\python.exe from C:\Users\thecesrom\AppData\Local\pypa\virtualenv\py_info\1\5acf275e529b7f1b869594580444d4303145808c0032da3f6f98a78be5318304.json [DEBUG via_disk_folder:129]
421 Attempting to release lock 2971795500784 on C:\Users\thecesrom\AppData\Local\pypa\virtualenv\py_info\1\5acf275e529b7f1b869594580444d4303145808c0032da3f6f98a78be5318304.lock [DEBUG _api:244]
422 Lock 2971795500784 released on C:\Users\thecesrom\AppData\Local\pypa\virtualenv\py_info\1\5acf275e529b7f1b869594580444d4303145808c0032da3f6f98a78be5318304.lock [DEBUG _api:247]
423 proposed PythonInfo(spec=CPython3.10.12.final.0-64, system=C:\Users\thecesrom\scoop\apps\coatl310\current\python.exe, exe=C:\Users\thecesrom\.virtualenv-20.21.1\Scripts\python.exe, platform=win32, version='3.10.12 (tags/v3.10.12:b4e48a444e, Jun  8 2023, 10:05:20) [MSC v.1936 64 bit (AMD64)]', encoding_fs_io=utf-8-utf-8) [INFO builtin:63]
423 accepted PythonInfo(spec=CPython3.10.12.final.0-64, system=C:\Users\thecesrom\scoop\apps\coatl310\current\python.exe, exe=C:\Users\thecesrom\.virtualenv-20.21.1\Scripts\python.exe, platform=win32, version='3.10.12 (tags/v3.10.12:b4e48a444e, Jun  8 2023, 10:05:20) [MSC v.1936 64 bit (AMD64)]', encoding_fs_io=utf-8-utf-8) [DEBUG builtin:65]
430 symlink on filesystem does work [DEBUG info:43]
431 filesystem is not case-sensitive [DEBUG info:24]
508 create virtual environment via CPython3Windows(dest=C:\Users\thecesrom\.virtualenv-virtualenv, clear=False, no_vcs_ignore=False, global=False) [INFO session:48]
Traceback (most recent call last):
  File "C:\Users\thecesrom\scoop\apps\coatl310\current\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\thecesrom\scoop\apps\coatl310\current\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\__main__.py", line 69, in <module>
    run_with_catch()  # pragma: no cov
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\__main__.py", line 54, in run_with_catch
    run(args, options, env)
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\__main__.py", line 16, in run
    session = cli_run(args, options, env)
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\run\__init__.py", line 30, in cli_run
    of_session.run()
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\run\session.py", line 42, in run
    self._create()
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\run\session.py", line 49, in _create
    self.creator.run()
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\create\creator.py", line 149, in run
    self.create()
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\create\via_global_ref\builtin\via_global_self_do.py", line 75, in create
    dirs = self.ensure_directories()
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\create\via_global_ref\builtin\via_global_self_do.py", line 100, in ensure_directories
    return {self.dest, self.bin_dir, self.script_dir, self.stdlib} | set(self.libs)
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\create\describe.py", line 44, in stdlib
    self._stdlib = Path(self.interpreter.sysconfig_path("stdlib", config_var=self._config_vars))
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\create\describe.py", line 56, in _config_vars
    self._conf_vars = self._calc_config_vars(self.dest)
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\create\describe.py", line 61, in _calc_config_vars
    return {k: (to if v.startswith(self.interpreter.prefix) else v) for k, v in sys_vars.items()}
  File "C:\Users\thecesrom\.virtualenv-20.21.1\lib\site-packages\virtualenv\create\describe.py", line 61, in <dictcomp>
    return {k: (to if v.startswith(self.interpreter.prefix) else v) for k, v in sys_vars.items()}
AttributeError: 'NoneType' object has no attribute 'startswith'
```

This renders `tox` completely useless on my machine.

**EDIT**
Ref: https://github.com/tox-dev/tox/issues/3018

**Comment by gaborbernat** [comment]
This version is no longer supported.

**Comment by thecesrom** [comment]
We run our CI on Ubuntu, and I also work on macOS, and both are unaffected by this problem, and a coworker is facing the same issue, so it's not exclusive to my machine.

So this begs the question, can `tox` only test end-of-life Python versions on Linux and macOS, but not on Windows?

**Comment by gaborbernat** [comment]
I don't think this is related to tox at all. This is an open source project and we do not support end of life. Pythons. You are free to put in a pull request if you want to fix something, however, don't expect either support or help with it besides reviewing and releasing that change.

**Comment by thecesrom** [comment]
The problem here is that when running virtualenv installed in a virtual environment is the root cause.

But what is odd is that tox with virtualenv<20.22 works fine on Linux and macOS not not on Windows. And I mention tox because it is how I found this issue, not because I blame tox.

I am not expecting anyone to support end-of-life tools/products, that's my responsibility within my organization. But considering that tox's FAQ suggests it still sports testing EOL Python, I was expecting it to work on Windows.

**Comment by thecesrom** [comment]
Perhaps using `virtualenv<20.21` instead of `virtualenv<20.22` will work on Windows. 

I'll give that a try and share my findings.

If successful, I might create a PR for tox's documentation.

**Comment by gaborbernat** [comment]
I don't think it's says It supports it, It says it might be able to get it working by applying that pin.

**Comment by thecesrom** [comment]
> Perhaps using `virtualenv<20.21` instead of `virtualenv<20.22` will work on Windows.

Unfortunately, this did not work.

Thanks for your assistance, @gaborbernat.

**Comment by thecesrom** [comment]
Just one last question.

Is it possible to restrict tox `requires` per platform, e.g. `sys_platform != 'win32'`, as one would in `deps`?

**Comment by thecesrom** [comment]
I have found a workaround by calling `nox` from `tox` and creating a `tox` configuration file for local tests.

Fun fact: `nox` with `virtualenv` pinned to version 20.21.1 works, but for some reason I couldn't do the same installing `tox==4.5.1` which requires `virtualenv>=20.21`.

Here's a sample of the configuration files:

`tox-dev.ini`:

```ini
[tox]
requires =
    tox>=4.2
env_list =
    install

[testenv:install]
description = test installation on Python 2.7.18 using nox
base_python = python3.10
skip_install = true
deps =
    argcomplete==3.0.8
    colorlog==6.7
    distlib==0.3.6
    filelock==3.12
    nox==2023.4.22
    packaging==23.1
    platformdirs==3.5.1
    virtualenv==20.21.1
commands =
    nox -e install
```

`noxfile.py`:

```python
"""Nox config."""

import nox


@nox.session(python="2.7")
def install(session: nox.Session) -> None:
    """Install package."""
    session.install(".")

```

`tox.ini` used for CI running on Ubuntu:

```ini
[tox]
requires =
    tox>=4.2
    virtualenv<20.22.0
env_list =
    install

[testenv:install]
description = test installation on Python 2.7.18
base_python = python2.7
```

**Comment by thecesrom** [comment]
The root cause, as stated before, is that `virtualenv` fails to create virtual environments when `virtualenv` is installed and running on a virtual environment.

Since I was installing `tox` using `pipx`, the solution was to install `tox` on the base Python.

If no further investigation is pending on your end, we can close this issue.

Thanks

**Comment by lord-haffi** [comment]
I just stumbled across this issue but with Python 3.12. My setup is a bit different (I will explain it in the following), but
TL;DR: I could fix it as well by upgrading the version of `virtualenv` in the corresponding venv.

I have a virtual env on my system called `.toxbase` with it's scripts path added to my path variable. Such that I can use tox from everywhere on my (Windows 11) system. This venv had `tox 4.12.1` and `virutalenv 20.25.0` installed.

I just cloned a new repository and built our dev environment through a tox call which also sets up `pre-commit`. But suddenly `pre-commit` threw this exact error. After some hours of trying (reinstalling etc.) I discovered that tox installed another version of `virtualenv` into the dev environment. This may had to do with a dependency in my project, idk. However, upgrading my dev environment to a newer version of `virtualenv` fixed my problem.

Just for saving some hours of headache for anybody encountering this issue as well :)
